### PR TITLE
Cleanup and docs

### PR DESCRIPTION
**Pull request by cmurphy**
* Remove unused packages and functions
* Cleanup minor code quality issues
* Fix HTTP compliance
* Add godoc comments for the stores/partition and stores/proxy packages. This is to set up a base for significant alterations to how the base stores work with pagination, sorting, and filtering.

Partially addresses https://github.com/rancher/rancher/issues/38593

### CODE REVIEW

**Review by rmweir — APPROVED** [review]
LGTM

**Review by MbolotSuse — CHANGES_REQUESTED** [review]
Thanks for adding comments to a lot of this stuff.

I have some concerns about changing exported functions/interfaces on a non-major release. I'd prefer we not do this (or that we tag the code after this is merged as v1.0.0) - as we are not the only users of steve.

One other note - your CI is going to fail until #59 goes in. If you want it to pass, you can update a specific part of the .drone.yml to fix that - see [here](https://github.com/rancher/steve/pull/59/files#diff-b54b39f1afced2465e1f3641db9d5bbf4f3a7fcf890996dfedd3c197bcb7f8c7R11-R16) for what you need to add.

**Comment by cmurphy** [comment]
@MbolotSuse I disagree about changing exported functions. steve does not follow semver. It does not even have releases. We don't make any guarantees of backwards compatibility. I've checked in harvester and throughout github and nothing is using these functions and packages. If I missed something, someone can file an issue and it is easy to revert. But I don't think trying to live up to an idealistic standard of API compatibility is helpful to us. It is more beneficial to clean up cruft because this helps us maintain the code in the long run.

File: pkg/server/router/router.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func Routes(h Handlers) http.Handler {
 	m.StrictSlash(true)
 	m.Use(urlbuilder.RedirectRewrite)
 
-	m.Path("/").Handler(h.APIRoot).HeadersRegexp("Accepts", ".*json.*")
+	m.Path("/").Handler(h.APIRoot).HeadersRegexp("Accept", ".*json.*")
```
[the remark this answers is not in the document]
Comment (cmurphy) [reply pkg/server/router/router.go:25]: https://github.com/rancher/dashboard/search?q=Accepts

**Comment by MbolotSuse** [comment]
Discussed offline - since Steve doesn't currently follow a versioning strategy, there's no incentive for us to preserve functionality between changes. Especially when the functionality isn't being used by other areas.

**Review by KevinJoiner — APPROVED** [review]
"[s]he is a hero, not the hero we deserved but the hero we needed"
🙇‍♂️

**Comment by cmurphy** [comment]
rebased

**Pull request merged**